nodl_python/nodl/_parsing/_parsing.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2020 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the
+# GNU Limited General Public License version 3, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranties of MERCHANTABILITY, SATISFACTORY QUALITY, or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Limited General Public License for more details.
+#
+# You should have received a copy of the GNU Limited General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+from typing import List, TYPE_CHECKING
+
+from lxml import etree
+from nodl._parsing import _v1 as parse_v1
+from nodl._parsing._schemas import get_schema
+from nodl.exception import InvalidNoDLError, UnsupportedInterfaceError
+from nodl.types import Node
+
+if TYPE_CHECKING:  # pragma: no cover
+    import pathlib
+
+
+NODL_MAX_SUPPORTED_VERSION = 1
+
+
+_SCHEMA = get_schema('interface.xsd')
+
+
+def _parse_interface(interface: etree._Element) -> List[Node]:
+    """Parse out all nodes from an interface element."""
+    if interface.get('version') == '1':
```

Comment:
lxml [explicitly recommends](https://lxml.de/api/lxml.etree._Element-class.html) using the accessor methods since dict creation is not guaranteed and can often be avoided for performance reasons. Not really a concern in our case, but using .get is technically the preferred way over accessing attrib and forcing a dict to be created.

Reply:
Yeah fair enough. Let's leave it as-is for now, but if keep this in mind if we end up hitting annoying issues.